src/main/java/ca/corefacility/bioinformatics/irida/ria/web/announcements/dto/AnnouncementUserReadDetails.java

```diff
@@ -0,0 +1,47 @@
+package ca.corefacility.bioinformatics.irida.ria.web.announcements.dto;
+
+import ca.corefacility.bioinformatics.irida.model.announcements.Announcement;
+import ca.corefacility.bioinformatics.irida.model.announcements.AnnouncementUserJoin;
+import com.google.common.collect.ComparisonChain;
+
+/**
+ * Data transfer object for retrieving announcement details from a sql query.
+ */
+
+public class AnnouncementUserReadDetails implements Comparable<AnnouncementUserReadDetails>{
+
+	private Announcement announcement;
+	private AnnouncementUserJoin announcementUserJoin;
+
+	public AnnouncementUserReadDetails(Announcement announcement, AnnouncementUserJoin announcementUserJoin) {
+		this.announcement = announcement;
+		this.announcementUserJoin = announcementUserJoin;
+	}
+
+	@Override
+	public int compareTo(AnnouncementUserReadDetails other) {
+		return ComparisonChain.start().compareTrueFirst(isRead(), other.isRead()).compareTrueFirst(announcement.getPriority(), other.getAnnouncement().getPriority()).compare(other.getAnnouncement().getCreatedDate(), announcement.getCreatedDate()).result();
+	}
+
+	public Announcement getAnnouncement() {
+		return announcement;
+	}
+
+	public void setAnnouncement(Announcement announcement) {
+		this.announcement = announcement;
+	}
+
+	public AnnouncementUserJoin getAnnouncementUserJoin() {
+		return announcementUserJoin;
+	}
+
+	public void setAnnouncementUserJoin(AnnouncementUserJoin announcementUserJoin) {
+		this.announcementUserJoin = announcementUserJoin;
+	}
+
+	public boolean isRead() {
```

Comment:
The isRead method is used for sorting. It was implemented when I was returning all announcements to the announcements page, but because I added the "All", "Read", and "Unread" tabs, you don't really notice this functionality anymore. Please see ced8bf9ad67f29abab1c3819d1b4c58f2bbe53b6.